core/templates/domain/exploration/InteractionCustomizationArgObjectFactory.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating an interaction customization argument.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { Schema } from 'services/schema-default-value.service';
+import {
+  SubtitledHtml, SubtitledHtmlObjectFactory, SubtitledHtmlBackendDict
+} from 'domain/exploration/SubtitledHtmlObjectFactory';
+import {
+  SubtitledUnicode, SubtitledUnicodeObjectFactory, SubtitledUnicodeBackendDict
+} from 'domain/exploration/SubtitledUnicodeObjectFactory';
+import {
+  InteractionCustomizationArgsBackendDictValue,
+  InteractionCustomizationArgsValue
+} from 'interactions/customization-args-defs';
+import { cloneDeep } from 'lodash';
+import { SchemaConstants } from
+  'components/forms/schema-based-editors/schema-constants';
+
+export interface InteractionCustomizationArgBackendDict {
+  value: InteractionCustomizationArgsBackendDictValue
+}
+
+export class InteractionCustomizationArg {
```

Comment:
Add docstrings to these classes, explain what they are. In particular be clear whether they represent a single customization arg (and, if so, what that concept actually means) or the entire set of customization args for an interaction.